examples/hitl/rearrange_v2/app_state_start_session.py

```diff
@@ -71,79 +79,171 @@ def _try_get_episode_ids(self) -> Optional[List[str]]:
+            if "episodes" not in connection_record:
+                print("Users did not request episodes. Cancelling session.")
+                return None
+            episodes_str = connection_record["episodes"]
 
-        # Validate that all users are requesting the same episodes.
-        for connection_record in data.connected_users.values():
-            if connection_record["episodes"] != episodes_str:
+            # Validate that the parameter is a string.
+            if episodes_str is None or not isinstance(episodes_str, str):
                 print(
-                    "Users are requesting different episodes! Cancelling session."
+                    f"Episodes are supplied in an unexpected format: '{type(episodes_str)}'. Cancelling session."
                 )
                 return None
 
-        # Validate that the episode set is not empty.
-        if episodes_str is None or len(episodes_str) == 0:
-            print("Users did not request episodes. Cancelling session.")
-            return None
+            # Parse episode string.
+            episodes_split = episodes_str.split(",")
+
+            for episode_str in episodes_split:
+                try:
+                    episodes[user_index].append(int(episode_str))
+                except Exception:
+                    print(
+                        f"Episode index '{episode_str}' is not an integer. Cancelling session."
+                    )
+                    return None
+
+            # Validate that the episode set is not empty.
+            if len(episodes[0]) == 0:
+                print("User did not request episodes. Cancelling session.")
+                return None
 
-        # Format: {lower_bound}-{upper_bound} E.g. 100-110
-        # Upper bound is exclusive.
-        episode_range_str = episodes_str.split("-")
-        if len(episode_range_str) != 2:
-            print("Invalid episode range. Cancelling session.")
-            return None
+            # Validate that all users are requesting the same episodes.
+            if user_index > 0 and episodes[user_index] != episodes[0]:
+                print(
+                    "Users are requesting different episodes. Cancelling session."
+                )
+                return None
 
-        # Validate that episodes are numeric.
-        start_episode_id = (
-            int(episode_range_str[0])
-            if episode_range_str[0].isdecimal()
-            else None
+        # Validate that the episodes are within range.
+        for episode in episodes[0]:
+            if episode < 0 or episode >= total_episode_count:
+                print(
+                    f"Episode index '{episode}' is out of bounds. There are '{total_episode_count}' episodes in this dataset. Cancelling session."
+                )
+                return None
+
+        return list(episodes[0])
+
+
+if __name__ == "__main__":
+    # TODO: Temporary test. Move to a dedicated test file.
```

Comment:
This is too messy to merge to `main`. You should go ahead and do this todo before merging.